plugins/dereferenceUses.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+const { querySelectorAll, querySelector } = require('../lib/xast.js');
+
+exports.type = 'full';
```

Comment:
Refactor with visitor please

Reply:
Sure! I am eager to try it with a plugin.